GitUI/UserControls/InteractiveGitActionControl.cs

```diff
@@ -0,0 +1,240 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GitCommands;
+using Microsoft.VisualStudio.Threading;
+using ResourceManager;
+
+namespace GitUI.UserControls
+{
+    public partial class InteractiveGitActionControl : GitModuleControl
+    {
+        private readonly TranslationString _progressMessage = new TranslationString("{0} is currently in progress.");
+        private readonly TranslationString _progressWithConflictsMessage = new TranslationString("{0} is currently in progress with merge conflicts.");
+
+        private readonly TranslationString _bisect = new TranslationString("Bisect");
+        private readonly TranslationString _rebase = new TranslationString("Rebase");
+        private readonly TranslationString _merge = new TranslationString("Merge");
+        private readonly TranslationString _patch = new TranslationString("Patch");
+
+        public enum State
+        {
+            None,
+            Bisect,
+            Rebase,
+            Merge,
+            Patch
+        }
+
+        private State _state = State.None;
+        private bool _hasConflicts = false;
+
+        public event EventHandler BisectEvent;
+
+        public InteractiveGitActionControl()
+        {
+            InitializeComponent();
+            InitializeComplete();
+        }
+
+        public async Task RefreshAsync()
+        {
+            // get the current state of the repo
+            await TaskScheduler.Default;
+
+            if (!Module.IsValidGitWorkingDir())
+            {
+                return;
+            }
+
+            if (Module.InTheMiddleOfBisect())
+            {
+                await SetStateAsync(State.Bisect, false);
+                return;
+            }
+
+            bool conflicted = Module.InTheMiddleOfConflictedMerge();
+
+            if (Module.InTheMiddleOfRebase())
+            {
+                await SetStateAsync(State.Rebase, conflicted);
+                return;
+            }
+
+            if (Module.InTheMiddleOfMerge())
+            {
+                await SetStateAsync(State.Merge, conflicted);
+                return;
+            }
+
+            if (Module.InTheMiddleOfPatch())
+            {
+                await SetStateAsync(State.Patch, conflicted);
+                return;
+            }
+
+            await SetStateAsync(State.None, false);
+        }
+
+        private async Task SetStateAsync(State state, bool conflicted)
```

Comment:
As above, there is nothin in this method that requires switching threads, we should be on UI thread here, so no `async` is necessary.